### ISSUE TITLE
textfiled will bounce when i drag into selected in bottom

### ISSUE THREAD

**Issue by linwenbang** [flutter/flutter#37377]
# Steps to Reproduce

![test](https://user-images.githubusercontent.com/11420716/62278551-e60afb80-b47a-11e9-974f-2f76f4daa0d0.gif)

1. set a textFiled, and set textStyle height 1.5
2. it will bounce when i drag into selected in bottom

## Logs

```
class TextPage extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return _TextPageState();
  }
}

class _TextPageState extends State<TextPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Text"),
      ),
      body: TextField(
        controller: TextEditingController(
            text:
                "撒地方加咖啡哈金凤凰卡号方式剪坏算法发挥福建垃asfjafjajfjaslfjaskjflasjfksajf撒分开建安路口附近拉设计费可使肌肤撒附近埃里克圾房卡设计费"),
        maxLines: null,
        style: TextStyle(color: Colors.red, height: 1.5),
        decoration: InputDecoration(
            border: InputBorder.none, contentPadding: EdgeInsets.all(1)),
      ),
    );
  }
}

```

```
[✓] Flutter (Channel unknown, v1.7.8+hotfix.4, on Mac OS X 10.14.3 18D42, locale zh-Hans)
    • Flutter version 1.7.8+hotfix.4 at /Users/linwenbang/development/flutter
    • Framework revision 20e59316b8 (13 days ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
    • Android SDK at /Users/linwenbang/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • CocoaPods version 1.7.5

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 36.0.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] VS Code (version 1.36.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.2.0

```

**Comment by justinmc** [comment]
I can reproduce this.  Any TextStyle height greater than the default causes it.

I spent awhile trying to track this down and couldn't yet pinpoint a cause.  Basically, this jumpTo is being called with a nonzero y offset, causing it to overscroll and bounce.

https://github.com/flutter/flutter/blob/5c147051352e1e9221ccdb9969195893abf57f54/packages/flutter/lib/src/widgets/editable_text.dart#L1642

We need to track that down and either discover a root cause or special case it for when there is no room to scroll.

**Comment by GaryQian** [comment]
It may be reasonable to ensure the caret never extends above or below the text itself. This should eliminate the bounce as the scrolling is determined through caret positioning.

**Comment by justinmc** [comment]
That would make sense to me.

As long as weird stuff like this still works:

<img width="287" alt="Screen Shot 2019-08-14 at 1 25 54 PM" src="https://user-images.githubusercontent.com/389558/63053683-3892e180-be97-11e9-8088-63f9c2e107e2.png">

`TextStyle(height: 10.0)`

**Comment by GaryQian** [comment]
Turns out that certain languages have extensions that go beyond the ascent metric. Cutting the caret short would cause it not cover the full glyph.

A better solution may be to simply not jump any further than the 0/bottom of the paragraph. Currently, we always jump to show the entire caret, but that may not make sense when the caret renders outside the formal definition of the paragraph, but is still valid text.

**Comment by GaryQian** [comment]
This should be fixed by #38573
